packages/beacon-node/src/util/blobs.ts

```diff
@@ -79,12 +79,15 @@ export async function dataColumnMatrixRecovery(
     return null;
   }
 
+  // Sort data columns by index in ascending order before passing for kzg operations
+  const partialSidecarsSorted = [...partialSidecars.values()].sort((a, b) => a.index - b.index);
```

Comment:
slight preference to do

```suggestion
  const partialSidecarsSorted = Array.from(partialSidecars.values()).sort((a, b) => a.index - b.index);
```

Reply:
+1

Reply:
Can move the sort down below to line 89 so it doesn't happen if all the columns are present